src/plan_expand_hypertable.c

```diff
@@ -115,6 +141,113 @@ const_datum_get_int(Const *cnst)
 	pg_unreachable();
 }
 
+/*
+ * Constify expressions of the following form in WHERE clause:
+ *
+ * column OP timestamptz - interval
+ * column OP timestamptz + interval
+ * column OP interval + timestamptz
+ *
+ * Iff interval has no month component.
+ *
+ * Since the operators for timestamptz OP interval are marked
+ * as stable they will not be constified during planning.
+ * However, intervals without a month component can be safely
+ * constified during planning as the result of those calculations
+ * does not depend on timezone settings.
+ */
+static OpExpr *
+constify_timestamptz_op_interval(PlannerInfo *root, OpExpr *constraint)
+{
+	Assert(constraint->args->length == 2);
```

Comment:
Compilers often complain about having code before declaration. This gets particularly weird with Asserts since they're only available in Debug builds. I would break this up to put the assertion after declarations.